sceptre/config/reader.py

```diff
@@ -141,6 +133,42 @@ def __init__(self, context):
 
         self.templating_vars = {"var": self.context.user_variables}
 
+        # Set up root
```

Comment:
I don't like the idea of the `__init__` doing this much. It's generally a bad pattern for a class' `__init__` method to actually _do_ things. It should only set the class up to do things. This sort of pattern makes the code harder to test, harder to maintain, and less flexible, because you simply cannot instantiate a ConfigReader without all this stuff happening. I'd rather the `__init__` do LESS rather than all this.